### ISSUE TITLE
Add or Update DateTime?

### ISSUE THREAD

**Issue by icolovic** [fo-dicom/fo-dicom#943]
### Expected behavior
AddOrUpdate function should not throw exception when value is a Nullable<DateTime>type but save the value to a tag.

### Actual behavior
Exception is thrown:
Unable to create DICOM element of type DT with values of type System.Nullable`1[System.DateTime]'

### Steps to reproduce the behavior
Call AddOrUpdate using Nullable<DiteTime> (DateTime?) as a value

### fo-dicom version and OS/platform
4.0.0 .NET Framework Desktop

**Comment by gofal** [comment]
Yes, currently the AddOrUpdate for tags with ValueRepresentation DT only work for `DateTime`, `DateRange` or `string`. it would be an enhancement to support nullabletypes everywhere.

You have to code:

    DateTime? nullDate;
    DicomDataset ds;

    // .... any code
    if (nullDate.HasValue)
       ds.AddOrUpdate(DicomTag.PatientBirthDate, nullDate.Value);
    else
        ds.AddOrUpdate(DicomTag.PatientBirthDate, string.Empty);

**Comment by gofal** [comment]
see #1922
